pkg/ccl/changefeedccl/changefeed_test.go

```diff
@@ -7420,6 +7421,20 @@ func TestChangefeedBackfillCheckpoint(t *testing.T) {
 		// Resume job.
 		require.NoError(t, jobFeed.Resume())
 
+		// Verify that the resumed job has restored the progress from the checkpoint
+		// to the change frontier.
+		knobs.FrontierSpan = func(frontier *resolvedspan.CoordinatorFrontier) {
+			require.NotNil(t, frontier)
+			var hasProgress bool
+			frontier.Entries(func(s roachpb.Span, ts hlc.Timestamp) (done span.OpResult) {
+				if !ts.IsEmpty() {
+					hasProgress = true
+				}
+				return span.ContinueMatch
+			})
+			require.Truef(t, hasProgress, "frontier should have saved checkpoint progress")
```

Comment:
Instead of just checking that we have some progress, we should assert that the span frontier matches what we expect (e.g. by constructing one from the checkpoint we have, maybe by using the `Restore` function)